File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int n = numbers.length;
        int left = 0, right = n - 1;
        while(left < right) {
            int sum = numbers[left] + numbers[right];
            if(left < right && sum < target) left++;
            else if (left < right && sum > target) right--;
            else return new int[] {left + 1, right + 1};
        }
        return new int[] {};
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        int n = nums.size();
        for(int i = 0; i < n; i++) {
            int needed = target - nums[i];
            if(map.find(needed) != map.end()) return {map[needed], i};
            map[nums[i]] = i;
        }
        return {-1, -1};
    }
};
